packages/common-sdk/src/web3/network/fetcher/simple-fetcher-impl.ts

```diff
@@ -0,0 +1,142 @@
+import { Connection } from "@solana/web3.js";
+import { AccountFetcher } from ".";
+import { Address, AddressUtil } from "../../address-util";
+import { getMultipleAccountsInMap } from "../account-requests";
+import { ParsableEntity } from "../parsing";
+
+type CachedContent<T> = {
+  parser: ParsableEntity<T>;
+  fetchedAt: number;
+  value: T | null;
+};
+
+export type RetentionPolicy<T> = ReadonlyMap<ParsableEntity<T>, number>;
+
+/**
+ * Options when fetching the accounts
+ */
+export type SimpleAccountFetchOptions = {
+  // Accepted maxAge in milliseconds for a cache entry hit for this account request.
+  maxAge?: number;
+};
+
+// SimpleAccountCache is a simple implementation of AccountCache that stores the fetched
+// accounts in memory. If TTL is not provided, it will use TTL defined in the the retention policy
+// for the parser. If that is also not provided, the request will always prefer the cache value.
+export class SimpleAccountFetcher<T, FetchOptions extends SimpleAccountFetchOptions>
+  implements AccountFetcher<T, FetchOptions>
+{
+  cache: Map<string, CachedContent<T>> = new Map();
+  constructor(readonly connection: Connection, readonly retentionPolicy: RetentionPolicy<T>) {
+    this.cache = new Map<string, CachedContent<T>>();
+  }
+
+  async getAccount<U extends T>(
+    address: Address,
+    parser: ParsableEntity<U>,
+    opts?: FetchOptions | undefined,
+    now: number = Date.now()
+  ): Promise<U | null> {
+    const addressKey = AddressUtil.toPubKey(address);
+    const addressStr = AddressUtil.toString(address);
+
+    const cached = this.cache.get(addressStr);
+    const maxAge = opts?.maxAge ?? this.retentionPolicy.get(parser) ?? Number.POSITIVE_INFINITY;
```

Comment:
To force refetch, I would be tempted to try and pass in `maxAge = 0`. I believe this line would end up using the retention policy, which could be confusing. Perhaps change to explicitly checking if `maxAge` is undefined here?

Reply:
Also the case where retentionPolicy has a zero value. Should that make the fetcher always fetch latest data?

Reply:
yeah good catch. upgraded the getMaxAge fn and added tests